x-pack/plugins/lens/public/app_plugin/lens_top_nav.tsx

```diff
@@ -337,6 +352,8 @@ export const LensTopNavMenu = ({
     application.capabilities,
   ]);
 
+  const lensStore = useStore();
```

Comment:
I noticed that you use `useStore` here and inside you the toggleSettingsMenuOpen you use it for dispatching the action and one selector: `const autoApplyEnabled = useLensSelector(selectAutoApplyEnabled);`. I had to read about it so maybe there's a reason for it, but wouldn't it be better to pass these two props (wrapped dispatch logic and selector) to `toggleSettingsMenuOpen` instead of passing the whole store? I'm basing this on this stackoverflow thread 😅 https://stackoverflow.com/questions/61292433/react-js-and-redux-useselector-vs-usestore 
Curious to learn if there's a reason this is better :)

Reply:
Makes sense! Thank you for detailed explanation 👌🏼